File: challenges/codeabbey/037/pach092.py
datos = raw_input().split()
cantidad = len(datos)
a = 0
p, i, m = int(datos[0]), (int(datos[1]) / 100.0) / 12, int(datos[2])
mensualidad = p*(i*(1+i)**m)/((1+i)**m-1)
print(round(mensualidad))

File: challenges/codeabbey/024/pach092.rb
cantidad = gets.chomp.to_i
datos = gets.chomp.split.map(&:to_i)
resultado = []
(0...cantidad).each do |a|
  contador = 0
  truncados = []
  numero = datos[a]
  until truncados.include? numero
    truncados << numero
    contador += 1
    numero = (numero * numero) / 100 % 10_000
  end
  resultado << contador
end
puts resultado.join(' ')
